test/clj-edn-el-parity-test.el

```diff
@@ -0,0 +1,286 @@
+
+(defun map-equal (m1 m2)
+  (and (and (hash-table-p m1) (hash-table-p m2))
+       (eq (hash-table-test m1) (hash-table-test m2))
+       (= (hash-table-count m1) (hash-table-count m2))
+       (equal (hash-table-keys m1) (hash-table-keys m2))
+       (equal (hash-table-values m1) (hash-table-values m2))))
+
+(defun make-seeded-hash-table (&rest keys-and-values)
+  (let ((m (make-hash-table :test #'equal)))
+    (while keys-and-values
+      (puthash (pop keys-and-values) (pop keys-and-values) m))
+    m))
+
+(ert-deftest maps ()
+  :tags '(edn maps)
+  (should (hash-table-p (clj-edn-read-str "{ }")))
+  (should (hash-table-p (clj-edn-read-str "{}")))
+  (should (map-equal (make-seeded-hash-table :foo :bar :baz :qux)
+                     (clj-edn-read-str "{ :foo :bar :baz :qux}")))
+  (should (map-equal (make-seeded-hash-table 1 "123" 'vector [1 2 3])
+                     (clj-edn-read-str "{ 1 \"123\" vector [1 2 3]}")))
+  (should (map-equal (make-seeded-hash-table [1 2 3] "some numbers")
+                     (clj-edn-read-str "{[1 2 3] \"some numbers\"}"))))
+
+(ert-deftest sets ()
+  :tags '(edn sets)
+  (should (eq 'edn-set (car (clj-edn-read-str "#{}"))))
+  (should (eq 'edn-set (car (clj-edn-read-str "#{ }"))))
+  (should (equal '(edn-set (1 2 3)) (clj-edn-read-str "#{1 2 3}")))
+  (should (equal '(edn-set (1 [1 2 3] 3)) (clj-edn-read-str "#{1 [1 2 3] 3}"))))
+
+(ert-deftest comment ()
+  :tags '(edn comments)
+  (should-not (clj-edn-read-str ";nada"))
+  (should (equal 1 (clj-edn-read-str ";; comment
+1")))
+  (should (equal [1 2 3] (clj-edn-read-str "[1 2 ;comment to eol
+3]")))
+  (should (equal '[valid more items] (clj-edn-read-str "[valid;touching trailing comment
+ more items]")))
+  (should (equal [valid vector more vector items] (clj-edn-read-str "[valid vector
+ ;;comment in vector
+ more vector items]"))))
+
+(defun test-val-passed-to-handler (val)
+  (should (listp val))
+  (should (= (length val) 2))
+  (should (= 1 (car val)))
+  1)
+
+(setq clj-edn-test-extra-handlers
+      (a-list
+       'my/type #'test-val-passed-to-handler
+       'my/other-type (lambda (val) 2)))
+
+(ert-deftest tags ()
+  :tags '(edn tags)
+  (should-error (clj-edn-read-str "#my/type value" clj-edn-test-extra-handlers))
+  (should (= 1 (clj-edn-read-str "#my/type (1 2)" clj-edn-test-extra-handlers)))
+  (should (= 2 (clj-edn-read-str "#my/other-type {:foo :bar}" clj-edn-test-extra-handlers))))
+
+(ert-deftest roundtrip ()
+  :tags '(edn roundtrip)
+  (let ((data [1 2 3 :foo (4 5) qux "quux"]))
+    (should (equal data (clj-edn-read-str (clj-edn-print-str data))))
+    (should (map-equal (make-seeded-hash-table :foo :bar)
+                       (clj-edn-read-str (clj-edn-print-str (make-seeded-hash-table :foo :bar)))))
+    (should (equal '(edn-set (1 2 3 [3 1.11]))
+                   (clj-edn-read-str (clj-edn-print-str '(edn-set (1 2 3 [3 1.11]))))))
+    (should-error (clj-edn-read-str "#myapp/Person {:first \"Fred\" :last \"Mertz\"}"))))
```

Comment:
good point, I guess edn'el had it here so I kept it there, but makes more sense to move it.